src/main.cpp

```diff
@@ -2187,450 +2158,69 @@ int main(int arg_count, char const **arg_ptr) {
-			defer (gb_string_free(lib_str));
-			char lib_str_buf[1024] = {0};
-
-			char const *output_ext = "exe";
-			gbString link_settings = gb_string_make_reserve(heap_allocator(), 256);
-			defer (gb_string_free(link_settings));
-
-
-			// NOTE(ic): It would be nice to extend this so that we could specify the Visual Studio version that we want instead of defaulting to the latest.
-			Find_Result_Utf8 find_result = find_visual_studio_and_windows_sdk_utf8();
-
-			if (find_result.windows_sdk_version == 0) {
-				gb_printf_err("Windows SDK not found.\n");
-				return 1;
-			}
-
-			if (build_context.ignore_microsoft_magic) {
-				find_result = {};
-			}
-
-			// Add library search paths.
-			if (find_result.vs_library_path.len > 0) {
-				GB_ASSERT(find_result.windows_sdk_um_library_path.len > 0);
-				GB_ASSERT(find_result.windows_sdk_ucrt_library_path.len > 0);
-
-				String path = {};
-				auto add_path = [&](String path) {
-					if (path[path.len-1] == '\\') {
-						path.len -= 1;
-					}
-					link_settings = gb_string_append_fmt(link_settings, " /LIBPATH:\"%.*s\"", LIT(path));
-				};
-				add_path(find_result.windows_sdk_um_library_path);
-				add_path(find_result.windows_sdk_ucrt_library_path);
-				add_path(find_result.vs_library_path);
-			}
-
-			for_array(i, ir_gen.module.foreign_library_paths) {
-				String lib = ir_gen.module.foreign_library_paths[i];
-				GB_ASSERT(lib.len < gb_count_of(lib_str_buf)-1);
-				isize len = gb_snprintf(lib_str_buf, gb_size_of(lib_str_buf),
-				                        " \"%.*s\"", LIT(lib));
-				lib_str = gb_string_appendc(lib_str, lib_str_buf);
-			}
-
-
-
-			if (build_context.build_mode == BuildMode_DynamicLibrary) {
-				output_ext = "dll";
-				link_settings = gb_string_append_fmt(link_settings, " /DLL");
-			} else {
-				link_settings = gb_string_append_fmt(link_settings, " /ENTRY:mainCRTStartup");
-			}
-
-			if (build_context.pdb_filepath != "") {
-				link_settings = gb_string_append_fmt(link_settings, " /PDB:%.*s", LIT(build_context.pdb_filepath));
-			}
-
-			if (build_context.no_crt) {
-				link_settings = gb_string_append_fmt(link_settings, " /nodefaultlib");
-			} else {
-				link_settings = gb_string_append_fmt(link_settings, " /defaultlib:libcmt");
-			}
-
-			if (ir_gen.module.generate_debug_info) {
-				link_settings = gb_string_append_fmt(link_settings, " /DEBUG");
-			}
-
-
-			char const *subsystem_str = build_context.use_subsystem_windows ? "WINDOWS" : "CONSOLE";
-
-			if (!build_context.use_lld) { // msvc
-				if (build_context.has_resource) {
-					i32 result = system_exec_command_line_app("msvc-link",
-						"\"rc.exe\" /nologo /fo \"%.*s.res\" \"%.*s.rc\"",
-						LIT(output_base),
-						LIT(build_context.resource_filepath)
-					);
-
-          if(result != 0) {
-            return 1;
-          }
-
-          result = system_exec_command_line_app("msvc-link",
-						"\"%.*slink.exe\" \"%.*s.obj\" \"%.*s.res\" -OUT:\"%.*s.%s\" %s "
```

Comment:
Where did the linker call go? Don't we still need that with the LLVM C API?

Reply:
The linker call is in a separate procedure. The old backend did everything inline.

Reply:
Ah fair enough

Reply:
See `linker_stage`